Source/Dafny/AST/DafnyAst.cs

```diff
@@ -8784,20 +8784,31 @@ public MatchStmt(IToken tok, IToken endTok, Expression source, [Captured] List<M
   public class MatchCaseStmt : MatchCase {
     private List<Statement> body;
     public Attributes Attributes;
+    // Has the case for this constructor been generated by the resolver because the pattern was
+    // a bound variable, or was it an explicit constructor case in the source code?
+    // E.g. 
+    // var x: Option<bool>;
+    // match x
+    //   case Some(true) => ... // FromBoundVar == false
+    //   case Some(_)    => ... // FromBoundVar == false
+    //   case v          => ... // FromBoundVar == true
+    //   case _ =>       => ... // FromBoundVar == true (this case would be unreachable; added for illustration purposes)
+    public readonly bool FromBoundVar;
```

Comment:
Is this variable true for all auto-generated cases, or just a subset of them?

Reply:
If you have
```
datatype X = A(x: X) | B(x: X) | C
match A(A(C))
  case A(A(x))) => 1
  case A(y) => 2
  case _ => 3 
```
then I believe Dafny will generate
```
datatype X = A(a: X) | B(b: X) | C
match A(A(C))
  case A(a) =>               // 1
    match a
      case A(a') =>          // 2
        match a'
          case A(a'') => 2   // 3
          case B(b'') => 2   // 4
          case C => 1        // 5
      case B(b) => 2         // 6
      case C => 2            // 7
  case B(b) => 3             // 8
  case C =>  3               // 9
```
I want to capture all the cases that capture `x` or `y`. So `FromBoundVar` should be true for (3-5) because those are captured by `x`, and (6-9) because those are captured by `y`.

Reply:
Essentially I'd like to capture all match cases that originate from a bound variable in the user's match case. Are you thinking of some specific example?

Reply:
Wonderful, thanks!